posts/comparing-quarkus-to-spring-through-the-eyes-of-jhipster.md

```diff
@@ -0,0 +1,120 @@
+
+## Queries
+### Spring Data
+Spring Data provides “syntactic sugar” for a variety of queries. Queries can be built simply by adding Repository methods with the correct naming scheme.
+```java
+List<User> findAllByActivatedIsFalseAndCreatedDateBefore(Instant dateTime)
+```
+And for more complex queries, the @Query annotation can be used.
+```java
+@Query(value = "select distinct ticket from Ticket ticket left join fetch ticket.labels")
+List<Ticket> findAllWithEagerRelationships();
+```
+### Quarkus
+Quarkus does not provide the method name query syntax. For simple add-ons to a typical list query, (i.e. an ORDER BY clause) the `PanacheEntityBase` methods provide String inputs.
+```java
+List<Person> persons = Person.list("order by name,birth");
+```
+More complex queries can be passed in this SQL-like syntax to the find and list methods.
+
+## Pagination
+### Spring Data
+With Spring Data, implement the `PagingAndSortingRepository` and a paginated `findAll` method is provided for the simple case.
+```java
+Page<T> findAll(Pageable pageable);
+```
+### Quarkus
+In Quarkus, the finder methods return a `PanacheQuery` object that can be operated on to provide pagination.
+```java
+// create a query for all living persons
+PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);
+
+// make it use pages of 25 entries at a time
+livingPersons.page(Page.ofSize(25));
+
+// get the first page
+List<Person> firstPage = livingPersons.list();
+
+// get the second page
+List<Person> secondPage = livingPersons.nextPage().list();
+```
+
+## REST Resources (Controllers)
+The REST resources are very similar between the two frameworks. The differences are basically syntactical and use about the same amount of code.
+
+This is an example class declaration:
+### Spring Data
+```java
+@RestController
+@RequestMapping("/api")
+public class AccountResource
+```
+### Quarkus
+```java
+@Path("/api")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@RequestScoped
+public class AccountResource
+```
+
+The method declarations are similar as well:
+### Spring Data
+```java
+@PostMapping("/register")
+@ResponseStatus(HttpStatus.CREATED)
+public void registerAccount(@Valid @RequestBody ManagedUserVM managedUserVM)
+```
+### Quarkus
+```java
+@POST
+@Path("/register")
+@PermitAll
+public CompletionStage<Response> registerAccount(@Valid ManagedUserVM managedUserVM)
+```
+
+# Conclusion
+It is clear that Spring is the more mature platform and has had many more years to provide some of the niceties that developers have grown used to having. More code is definitely required to use Quarkus at this point. There are plenty of resources online that talk about performance gains from using Quarkus so that very well could drive the decision to try it out. Based on this introductory analysis, the learning curve does not look too steep for Quarkus. I think the next step could be to investigate how Quarkus handles security and other details.
```

Comment:
```suggestion
It is clear that Spring is the more mature platform and has had many more years to provide some of the ergonomics that developers have grown used to having. More code is definitely required to use Quarkus at this point. There are plenty of resources online that talk about performance gains from using Quarkus so that very well could drive the decision to try it out. Based on this introductory analysis, the learning curve does not look too steep for Quarkus. A future blog post will investigate how Quarkus handles security and other developer concerns.
```